git_machete/github.py

```diff
@@ -23,11 +25,65 @@ def __repr__(self) -> str:
 
 GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
 
+# GitHub Enterprise deployments use alternate domains.
+# The logic in this module will need to be expanded to detect
+# and use alternate remote domains to provide enterprise support.
+GITHUB_DOMAIN = "github.com"
+GITHUB_REMOTE_PATTERNS = [
+    "^https://github\\.com/(.*)/(.*)\\.git$",
+    "^git@github\\.com:(.*)/(.*)\\.git$",
+]
 
-def github_token() -> Optional[str]:
+
+def _token_from_gh() -> Optional[str]:
+
+    # Abort without error if `gh` isn't available
+    gh = shutil.which('gh')
+    if not gh:
+        return None
+
+    # Run via subprocess.run as we're insensitive to return code.
+    #
+    # `gh` can store auth token for public and enterprise domains, specify
+    # single domain for lookup.
+    # This is *only* github.com until enterprise support is added.
+    # https://github.com/VirtusLab/git-machete/issues/137
+    proc = subprocess.run(
+        [gh, "auth", "status", "--hostname", GITHUB_DOMAIN, "--show-token"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    # gh auth status outputs to stderr in the form:
+    #
+    # {domain}:
+    #   ✓ Logged in to {domain} as {username} ({config_path})
+    #   ✓ Git operations for {domain} configured to use {protocol} protocol.
+    #   ✓ Token: *******************
+    #
+    # with non-zero exit code on failure
+    result = proc.stderr.decode()
+
+    match = re.search(r"Token: (\w+)", result)
+    if match:
+        return match.groups()[0]
+
+    return None
+
+
+def _token_from_env() -> Optional[str]:
     return os.environ.get(GITHUB_TOKEN_ENV_VAR)
 
 
+def github_token() -> Optional[str]:
+    token = _token_from_env()
+
+    if not token:
+        token = _token_from_gh()
```

Comment:
Actually! would it also make sense to check the token used by the older/unofficial GitHub CLI, `hub`? It's stored under `~/.config/hub` file:

```yaml
github.com:
- user: PawelLipski
  oauth_token: .........
  protocol: https
```